core/trino-main/src/main/java/io/trino/operator/join/unspilled/JoinProbe.java

```diff
@@ -83,7 +105,7 @@ public boolean isFinished()
 
     public long getCurrentJoinPosition()
     {
-        return joinPositionCache[position];
+        return joinPositionCache[position & rleMask];
```

Comment:
This maybe be almost free, but still we are making the non-RLE common case slightly more expensive now (unless there is some JIT magic to optimize this away at runtime).
Why don't we approach this as writing a `processRleProbe` code path in PageJoiner instead ?

Reply:
The & operator is usually a single cycle and executed by a different ALU that the "main" execution path so this should be actually free. At least that is my hypothesis.
We can rewrite `PageJoiner`, I see no downsides in that. But this is beyond my resources so someone else would need to do it.